helix-core/src/main/java/org/apache/helix/manager/zk/ZkBaseDataAccessor.java

```diff
@@ -1248,4 +1240,123 @@ public void close() {
       _zkClient.close();
     }
   }
+
+  public static class Builder {
+    private String zkAddress;
+    private RealmAwareZkClient.RealmMode realmMode;
+    private RealmAwareZkClient.RealmAwareZkConnectionConfig realmAwareZkConnectionConfig;
+    private RealmAwareZkClient.RealmAwareZkClientConfig realmAwareZkClientConfig;
+
```

Comment:
We'd need ZkBaseDataAccessor-specific parameters here as well such as ZkClientType (See https://github.com/apache/helix/pull/863 for reference)